### DISCUSSION TITLE
Build and deploy Docker, without deploying?

### DISCUSSION THREAD

**Discussion by alexander-potemkin** [caprover/caprover#2085] [Ideas]
Hi! Thank you for the wonderful software!

Is there any way how I can get CapRover triggered (could be manual) to build Dockerfile and push it to the self-hosted registry, but without immediate / automatic deployment?

Use case: get all code pushed to specific git branch build and ready to be deployed, but only be deployed manually.

Happy to do it via manual API calls if required, would appreciate any hints on that!

**Comment by githubsaturn** [comment]
The best workaround is to create a dummy app. Then go to app config and set the instance count to "0". The you set automatic deployment via git. But, because you have 0 instances, nothing gets deployed automatically. When you want to manually deploy, you can go to the original app and deploy the respective image that you want to deploy.

**Reply by alexander-potemkin** [reply]
Thank you, that's quite elegant!
Will it be pushed to the internal Docker registry as well?

**Reply by githubsaturn** [reply]
Yes.

**Reply by alexander-potemkin** [reply]
Thank you!
